src/Symfony/Component/HttpKernel/Config/FileLocator.php

```diff
@@ -35,6 +35,7 @@ public function __construct(KernelInterface $kernel, string $path = null, array
         if (null !== $path) {
             $this->path = $path;
             $paths[] = $path;
+            @trigger_error('Using the global fallback to load resources is deprecated since Symfony 4.4 and will be removed in Symfony 5.0.', E_USER_DEPRECATED);
```

Comment:
That's not enough: what will we do with the argument in 5.0?

Reply:
I'm making it `@final` So in 5.0 we can safely remove the argument.

Reply:
We cannot change the order of arguments. Are we going to deprecate them both?

Reply:
No we don't, we only need to deprecate $path, so the only solution I can see is to deprecate the whole class.

Reply:
Sorry if I wasn't clear enough, here is what I meant:
the patched method is the constructor, so making the class final won't help and should be removed
then, the signature should be changed now to:
`(KernelInterface $kernel, $paths = [])`
then, the implementation should deal with BC, using type checks and func_get_arg or similar, as we do usually (see existing code for some clues)

Reply:
done